pyart/io/output_to_geotiff.py

```diff
@@ -0,0 +1,246 @@
+from __future__ import print_function, division
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import os
+try:
+    from osgeo import gdal
+    IMPORT_FLAG = True
+except ImportError:
+    IMPORT_FLAG = False
+
+
+def write_grid_geotiff(filename, grid, field=None, rgb=False, level=None,
+                       cmap='viridis', vmin=0, vmax=75, color_levels=None,
+                       warp=False, sld=False):
+    """
+    Write a Py-ART Grid object to a GeoTIFF file. The GeoTIFF can be the
+    standard Azimuthal Equidistant projection used in Py-ART, or a lat/lon
+    projection on a WGS84 sphere. The latter is typically more usable in web
+    mapping applications. The GeoTIFF can contain a single float-point raster
+    band, or three RGB byte raster bands. The former will require an SLD file
+    for colorful display using standard GIS or web mapping software, while the
+    latter will show colors "out-of-the-box" but lack actual data values. The
+    function also can output an SLD file based on the user-specified inputs.
+    User can specify the 2D vertical level to be output. If this is not
+    specified, a 2D composite is created. User also can specify the field
+    to output.
+
+    This function requires GDAL Python libraries to be installed. These are
+    available via conda; e.g., 'conda install gdal'
+
+    Parameters
+    ----------
+    filename : str
+        Filename to save grid to
```

Comment:
Now refer to `filename` argument as a filename template. The function also can now handle it if the user did not add a suffix.